frontend/libs/studio-components/src/components/StudioCodelistEditor/StudioCodeListEditor.test.tsx

```diff
@@ -264,6 +242,42 @@ describe('StudioCodeListEditor', () => {
     await user.type(validValueInput, newValue);
     expect(onInvalid).not.toHaveBeenCalled();
   });
+
+  it('Should save all values as numbers, when a code list that contains numbers is modified', async () => {
+    const user = userEvent.setup();
+    renderCodeListEditor({ codeList: codeListWithNumbers });
+
+    const valueInput = screen.getByRole('textbox', { name: texts.itemValue(1) });
+    const newValue = '3.1416';
+    await user.type(valueInput, newValue);
+
+    expect(onChange).toHaveBeenCalledTimes(newValue.length);
+    expect(onChange).toHaveBeenLastCalledWith([
+      { ...codeListWithNumbers[0], value: Number(newValue) },
+      codeListWithNumbers[1],
+      codeListWithNumbers[2],
+    ]);
+  });
```

Comment:
Jeg prøvde å modifisere denne testen til å teste en kodeliste med kun ett element. Da feiler den fordi verdien blir konvertert til en tekststreng.

Hvis jeg endrer verdien til `'3,1416'` (med komma), feiler testen også. Vi bør nok ta høyde for komma siden det er slik vi skriver desimaltall på norsk.